src/main/java/org/opensearch/performanceanalyzer/reader_writer_shared/EventLogFileHandler.java

```diff
@@ -113,11 +115,18 @@ public void renameFromTmp(long epoch) {
     public void renameFromTmpWithPrivilege(long epoch) {
         Path path = Paths.get(metricsLocation, String.valueOf(epoch));
         Path tmpPath = Paths.get(path.toString() + TMP_FILE_EXT);
-        // This is done only when no exception is thrown.
-        try {
-            Files.move(tmpPath, path, REPLACE_EXISTING, ATOMIC_MOVE);
-        } catch (IOException e) {
-            LOG.error("Error moving file {} to {}.", tmpPath.toString(), path.toString(), e);
+        File tempFile = new File(tmpPath.toString());
+        // Only if the tmp file is present, we want to create the writer file.
+        // If not, we will publish a metric.
+        if (tempFile.exists()) {
+            // This is done only when no exception is thrown.
+            try {
+                Files.move(tmpPath, path, REPLACE_EXISTING, ATOMIC_MOVE);
+            } catch (IOException e) {
+                LOG.error("Error moving file {} to {}.", tmpPath.toString(), path.toString(), e);
+            }
```

Comment:
From our discussion, the reason the check `tempFile.exists()` is added : `Files.move` in case `source` is not present, creates an empty file with the name `source`. This will lead to empty .tmp files created, which will not be rotated as part of `writeAndRotate()`

I think we can avoid the check and rely on `MetricsPurgeActivity` to eventually delete these files.

Reply:
Re-visiting the above comment :


1/ we do not create `.tmp` file and we do not rotate the previous `.tmp` file from #0 window
2/ As part of next rotation, we look for `.tmp file` from #1 window, the file will not be present and we will create a new rotated file with `File.move()`
Now, the .tmp file from #0 will be cleaned by PurgeActivity but the rotated and renamed file from #1 will be an empty file, which will not solve our problem. Thus we need the .exists()  check

Reply:
Quick clarification here on `Files.move` behavior. When the source file is not present, then it throws IOException. Retaining this check here as it provides better readability and ability to emit a metric when the file is not present.